presto-main/src/main/java/io/prestosql/execution/executor/TaskExecutor.java

```diff
@@ -512,7 +512,9 @@ public void run()
                         if (!split.isDestroyed()) {
                             if (t instanceof PrestoException) {
                                 PrestoException e = (PrestoException) t;
-                                log.error("Error processing %s: %s: %s", split.getInfo(), e.getErrorCode().getName(), e.getMessage());
+                                String message = format("Error processing %s: %s: %s", split.getInfo(), e.getErrorCode().getName(), e.getMessage());
+                                log.error(message);
+                                log.debug(e, message);
```

Comment:
I'd do this instead to void double logging:
```java
if (log.isDebugEnabled()) {
    log.debug(e, message);
} 
else {
    log.error(message);
}
```

Reply:
Enabling DEBUG should not "hide" ERROR-level logging.
For example, I can run with DEBUG enabled and have downstream monitoring which eg considers ERROR/s rate

Reply:
Ah good point.  I think you want this instead:
```java
if (log.isDebugEnabled()) {
    log.error(e, message);
} 
else {
    log.error(message);
}
```

Reply:
Then every log message will go to ERROR. 
It's natural to expect that raising log level to INFO, WARN or ERROR won't eliminate any information, but it will.

Reply:
In fact i'd propose something else:
```
log.error(message + " (full stacktrace logged on DEBUG)");
log.debug(e, message);
```

I've done so in the past and so some lib doing so.